bgpd/bgp_route.c

```diff
@@ -414,6 +414,29 @@ bgp_info_cmp (struct bgp *bgp, struct bgp_info *new, struct bgp_info *exist,
   newattre = newattr->extra;
   existattre = existattr->extra;
 
+  if (exist->sub_type == BGP_ROUTE_NORMAL
+      && new->sub_type != BGP_ROUTE_NORMAL
+      && newattr->distance != 0) {
+    struct prefix *exist_prefix = &(exist->net->p);
+    u_char bgp_distance = bgp_distance_apply(exist_prefix, exist,
+          exist_prefix->family, SAFI_UNICAST, bgp);
+    if (newattr->distance < bgp_distance) {
+      return -1;
+    } else if (bgp_distance < newattr->distance) {
+      return 1;
+    }
+  } else if (new->sub_type == BGP_ROUTE_NORMAL
+             && exist->sub_type != BGP_ROUTE_NORMAL
+             && existattr->distance != 0) {
+    struct prefix *new_prefix = &(new->net->p);
+    u_char bgp_distance = bgp_distance_apply(new_prefix, new,
+          new_prefix->family, SAFI_UNICAST, bgp);
+    if (existattr->distance < bgp_distance) {
```

Comment:
Why is admin distance being looked at for bestpath calculation?  The RFCs do not say anything about using admin distance for bestpath calc.

Reply:
@dwalton76 @riw777 I am sorry for the delay in responding. Thank you for reviewing the patch. As @riw777 pointed out, when the distance on a static route is changed, we need the distance to be compared with BGP route's distance and bestpath selected based on distance. Currently, if there is a static route, BGP does not even push the dynamic route to Zebra irrespective of distance.